bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,356 @@
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.VOLTAGE_L1;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.VOLTAGE_L2;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.VOLTAGE_L3;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.CURRENT_L1;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.CURRENT_L2;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.CURRENT_L3;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.POWER_L1;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.POWER_L2;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.POWER_L3;
+import static org.openhab.binding.goecharger.internal.GoEChargerBindingConstants.ACCESS_STATE;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.goecharger.internal.GoEChargerBindingConstants;
+import org.openhab.binding.goecharger.internal.GoEChargerConfiguration;
+import org.openhab.binding.goecharger.internal.api.GoEStatusResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+
+/**
+ * The {@link GoEChargerHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Samuel Brucksch - Initial contribution
+ */
+@NonNullByDefault
+public class GoEChargerHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(GoEChargerHandler.class);
+
+    private @Nullable GoEChargerConfiguration config;
+
+    private Gson gson;
+    private @Nullable GoEStatusResponse goeResponse;
```

Comment:
I don't think you need this. Just pass response data from `GetGoEData` to other methods as a parameter.